### PR TITLE
Make it possible to run OSEM on heroku

### PR DESCRIPTION
**Pull request by hennevogel**
Next up: Making file upload work. I'm contemplating of using https://github.com/cloudinary/cloudinary_gem for this

### CODE REVIEW

File: app/controllers/conference_registrations_controller.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def new
       redirect_to edit_conference_conference_registrations_path(@conference.short_title)
       return
     # ichain does not allow us to create users during registration
-    elsif CONFIG['authentication']['ichain']['enabled'] && !current_user
+    elsif (ENV['OSEM_ICHAIN_ENABLED'] == 'true') && !current_user
```
Comment (ChrisBr) [inline app/controllers/conference_registrations_controller.rb]: Hm, is there no way to parse the ENV variable into a boolean?

File: app/models/email_settings.rb [hunk 2]
```diff
@@ -7,12 +7,12 @@ def get_values(conference, user, event = nil)
       'conference_start_date' => conference.start_date,
       'conference_end_date' => conference.end_date,
       'registrationlink' => Rails.application.routes.url_helpers.conference_conference_registrations_url(
-                            conference.short_title, host: CONFIG['url_for_emails']),
+                            conference.short_title, host: (ENV['OSEM_HOSTNAME'] || 'localhost:3000')),
```
Comment (ChrisBr) [inline app/models/email_settings.rb]: Actually I don't like to hardcode ``(ENV['OSEM_HOSTNAME'] || 'localhost:3000')``
(same for all the others)

File: app/views/conference_registrations/_form.html.haml [hunk 3]
```diff
@@ -11,7 +11,7 @@
         %legend
           %span
             =link_to('#signup', role: 'tab', "aria-controls" => "home", "data-toggle" => "tab") do
-              = CONFIG['name']
+              = ENV['OSEM_NAME'] || 'OSEM'
```
Comment (ChrisBr) [inline app/views/conference_registrations/_form.html.haml]: And I also don't like to hardcode OSEM on various places ...

**Comment by ChrisBr** [comment]
@hennevogel how does setting in the env variables in vagrant work? Works it automatically (haven't seen code for it ...) or do I need to set it manually (I wouldn't like it ...)

File: app/controllers/conference_registrations_controller.rb (hunk 1 above)
Comment (hennevogel) [inline app/controllers/conference_registrations_controller.rb]: No it's always a string. Usually this is done by existance. Either it is set, no matter the content, or not. Would you like that more?

File: app/models/email_settings.rb (hunk 2 above)
Comment (hennevogel) [inline app/models/email_settings.rb]: Then set the variable :-)

**Comment by hennevogel** [comment]
You don't have to set any env variable in vagrant...

**Comment by sonalkr132** [comment]
> I'm contemplating of using https://github.com/cloudinary/cloudinary_gem for this

:+1: It is not that difficult to set up. Users would just need to copy their keys to `.cloudinary.yml` file. Free account features:
75,000 Total Images & Videos
7,500 Monthly Transformations
2 GB Managed Storage
5 GB Monthly Net Viewing Bandwidth

We should also provide some option for our background job. Workers on herkou have limited free hours. Here are possible alternatives:
* `rake jobs:workoff` with [Heroku Scheduler](https://devcenter.heroku.com/articles/scheduler). As you can guess, this will suffer with significant delay.
* [Workless](https://github.com/lostboy/workless) gem
* We can shift from delayed_jobs to redis :heart: , heroku has a reasonable size (for our purpose) free addon for this (Chekour: [Redis To Go](https://elements.heroku.com/addons/redistogo) )

**Comment by ChrisBr** [comment]
@hennevogel please rebase!

**Comment by ChrisBr** [comment]
Have you tried if it still works with vagrant?
(with a new box and with an existing box?)

**Comment by hennevogel** [comment]
Nope will do

**Comment by hennevogel** [comment]
Tested with a newly provisioned vagrant box, everything works as expected

**Comment by hennevogel** [comment]
@ChrisBr now there is a dotenv example file that you can copy to set variables

**Comment by hennevogel** [comment]
cloudinary is done in #970

File: config/secrets.yml.example [hunk 4]
```diff
@@ -1,18 +1,9 @@
...
-  # Your errbit API key
-  # errbit_key: '12345'
-
 development:
-  <<: *defaults
-  # Sample data so that mocks work
+  # Generate your own with rake secret
+  # secret_key_base: '12345'
+
+  ########## OMNIAUT Providers ##########
```
Comment (differentreality) [inline config/secrets.yml.example]: typo: OMNIAUTH

File: config/secrets.yml.example [hunk 5]
```diff
@@ -23,33 +14,45 @@ development:
...
-  <<: *defaults
-  # Leave the variables' names empty, unless you use the providers, in which case you need to
-  # register your applicaton and add the actual keys.
+  # Generate your own with rake secret or use the environment
+  # secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
+
+  # Your errbit API key
+  # errbit_key: '12345'
+
+  ########## OMNIAUT Providers ##########
```
Comment (differentreality) [inline config/secrets.yml.example]: typo: OMNIAUTH

File: config/environment.rb [hunk 6]
```diff
@@ -3,11 +3,18 @@
...
-  puts "Error while parsing config file #{path}"
-  CONFIG = {}
+if File.exist?(path)
+  puts "
+
+WARNING: The OSEM configuration file
+
+#{path}
+
+is deprecated. Please use the environment environment variables
```
Comment (differentreality) [inline config/environment.rb]: one environment enough?

**Comment by hennevogel** [comment]
@differentreality fixed the typos

**Comment by differentreality** [comment]
I havent tried it with vagrant, other than that everything looks good :+1:

**Pull request merged**